bag_register/src/main/scala/uk/ac/wellcome/platform/archive/bag_register/services/BagRegisterWorker.scala

```diff
@@ -44,15 +54,23 @@ class BagRegisterWorker[IngestDestination, OutgoingDestination](
         space = payload.storageSpace
       )
 
-      _ <- ingestUpdater.send(
-        ingestId = payload.ingestId,
-        step = registrationSummary
-      )
-
-      _ <- outgoingPublisher.sendIfSuccessful(
-        result = registrationSummary,
-        outgoing = payload
-      )
+      _ <- Future.fromTry {
+        ingestUpdater.send(
+          ingestId = payload.ingestId,
+          step = registrationSummary
+        )
+      }
 
+      _ <- Future.fromTry {
+        outgoingPublisher.sendIfSuccessful(
+          result = registrationSummary,
+          outgoing = payload
+        )
+      }
     } yield registrationSummary
+
+  override def processMessage(
```

Comment:
Why aren't we using this?

Reply:
I’ll drop in a comment.

Reply:
See comment in https://github.com/wellcomecollection/storage-service/pull/543/commits/c7c725aa9e5029bda8185d8a94037bf7c5695936

Reply:
Perhaps this is a situation where we should turn a future into a try by waiting for it rather than circumvent `processMessage` - we're waiting for an internal REST API to return, maybe a hard time limit here makes sense.

Reply:
Would certainly simplify some of the internal APIs if we could Try instead of Future. I wonder if that should be pushed into the tracker client? We don't really use Future anywhere else; it's bits of Akka that are leaking.

Reply:
I think there should be an explicit "amount of time to wait for the tracker api to return somewhere" at your discretion where that it.

Reply:
Probably the tracker client? Say, the tracker API has to return within 20 seconds or we give up?